skipruntime-ts/core/src/skipruntime_api.ts

```diff
@@ -69,6 +72,31 @@ export abstract class ValueMapper<
   }
 }
 
+/**
+ * A specialized form of `Mapper` which maps values to values, reusing the
+ * input collection's key structure in the output collection. Use this form
+ * when the output values for a particular key need to consider all the
+ * input values associated to the key.
```

Comment:
The last part ("when the output values for a particular key need to consider all the input values associated to the key") is a bit unclear, but maybe it's just me. Maybe stress that the output values do *not* need to consider input values for other keys?

Reply:
Actually scratch that, I was confused. I guess the confusing part of the sentence is that it suggests the alternative is for the output values to need to consider only _some_ of the input values associated to the key. If this is the case, maybe mention something like ("unlike with a FooMapper that requires only some").

Reply:
Hmm, I sense that there is still some confusion here, and digging in is probably worthwhile as users will likely hit it too. The crucial distinction is that one type of mapper is meant for the case where when a collection has multiple values for a key it is just a representation for a set of key-value pairs where the keys need not be unique, and the other type of mapper is meant for the case where the value associated with a key is already a (mathematical) set. So if you have a collection mapping keys to numbers, then the second type of mapper can be used to transform a <key>-<numbers> to <key>-<the sum of the numbers associated to the key>, while the first type of mapper cannot. I need to find a better explanation.